### ISSUE TITLE
`outdated` & `upgrade` get stuck downgrading macOS app

### ISSUE THREAD

**Issue by emilazy** [mas-cli/mas#335]
## Your Environment

- `mas version`: 1.8.1
- macOS version (`system_profiler SPSoftwareDataType -detailLevel mini`): 11.2.3

### mas Install Method

`brew install mas` (homebrew-core)

## Describe the Bug

`mas` wants to constantly update an app that isn’t out of date.

## To Reproduce

Install DaisyDisk and try `mas upgrade`/`mas outdated`. It’s possible this is related to some quirk of my environment, but I can’t imagine what. I don't think this is related to #290 because there’s no iOS version of DaisyDisk.

## Expected Behavior

`mas update` doesn’t try to reinstall the same version over and over.

## Actual Behavior

`mas update` tries to reinstall the version over and over.

## Screenshots, Terminal Output

```console
$ mas upgrade
Upgrading 1 outdated application:
DaisyDisk (4.12.1)
==> Downloading DaisyDisk
==> Installed DaisyDisk
$ mas upgrade
Upgrading 1 outdated application:
DaisyDisk (4.12.1)
==> Downloading DaisyDisk
==> Installed DaisyDisk
$ mas outdated
411643860 DaisyDisk (4.12.1 -> 4.12)
```

**Comment by phatblat** [comment]
Thanks for the bug report! We've seen a lot of inconsistencies with versions since these commands were restored in 1.8.0. I get the same message about DaisyDisk.

**Comment by chris-araman** [comment]
This should be resolved by #346.

**Comment by emilazy** [comment]
Thank you very much for fixing this! I just built e51366ce90a97544db0106347fd661ff88041b22 and can confirm that `mas outdated` is now correctly empty for me. Looking forward to seeing a release with #346 in it!
